src/app/clusters/resource-monitoring-server/resource-monitoring-cluster-objects.h

```diff
@@ -141,37 +141,39 @@ struct TypeInfo
 };
 } // namespace LastChangedTime
 
-namespace ReplacementProductStruct {
+namespace GenericReplacementProductStruct {
 enum class Fields : uint8_t
 {
     kProductIdentifierType  = 0,
     kProductIdentifierValue = 1,
 };
 
-struct Type
+class GenericType : protected app::Clusters::HepaFilterMonitoring::Structs::ReplacementProductStruct::Type
 {
-public:
-    ProductIdentifierTypeEnum productIdentifierType = static_cast<ProductIdentifierTypeEnum>(0);
+private:
+    ResourceMonitoring::ProductIdentifierTypeEnum productIdentifierType;
     chip::CharSpan productIdentifierValue;
 
-    CHIP_ERROR Decode(TLV::TLVReader & reader);
-
+public:
+    GenericType() {}
+    GenericType(ResourceMonitoring::ProductIdentifierTypeEnum aProductIdentifierType, chip::CharSpan aProductIdentifierValue)
+    {
+        productIdentifierType  = aProductIdentifierType;
+        productIdentifierValue = aProductIdentifierValue;
+    }
+    virtual ~GenericType()                = default;
     static constexpr bool kIsFabricScoped = false;
-
     CHIP_ERROR Encode(TLV::TLVWriter & writer, TLV::Tag tag) const;
 };
-
-using DecodableType = Type;
-
-} // namespace ReplacementProductStruct
+} // namespace GenericReplacementProductStruct
```

Comment:
This struct needs to have a buffer for the character data the CharSpan is pointed too.  It should probably have setters for the type and value; type assigns to the type member, value (a fallible setter!) copies the data into the internal buffer and points the charspan to that buffer.

In particular, this PR in its current form does _not_ address https://github.com/project-chip/connectedhomeip/issues/28148.

I would strongly urge you to exercise this logic by having this attribute return _not_ static data in some cases.  And add tests!

Reply:
Modified this and also added a test for it